### ISSUE TITLE
Support for PHP 8.4 in Vimeo Psalm

### ISSUE THREAD

**Issue by sandunMadhushan** [vimeo/psalm#11194]
I am encountering an issue while trying to use vimeo/psalm with PHP 8.4.1. It appears that the package currently supports PHP versions up to 8.3.x, but does not include support for PHP 8.4.

Here are the details of the issue:

**Problem**
When installing vimeo/psalm, I get the following error:

- Root composer.json requires vimeo/psalm ^5 -> satisfiable by vimeo/psalm[5.0.0, ..., 5.26.1].
- vimeo/psalm[5.0.0, ..., 5.13.1] require php ^7.4 || ~8.0.0 || ~8.1.0 || ~8.2.0 -> your php version (8.4.1) does not satisfy that requirement.
- vimeo/psalm[5.14.0, ..., 5.26.1] require php ^7.4 || ~8.0.0 || ~8.1.0 || ~8.2.0 || ~8.3.0 -> your php version (8.4.1) does not satisfy that requirement.

Are there plans to add support for PHP 8.4?
If so, is there a timeline for when this might be implemented?

**Environment**
PHP Version: 8.4.1
Composer Version: 2.8.4
Vimeo Psalm Version: 5.x.x (latest)

**Comment by danog** [comment]
Please require `dev-master`.

**Comment by michaelcozzolino** [comment]
@danog will there be a release for php8.4 support?

**Comment by zmitic** [comment]
> once I get started on my current support contract for Psalm

@danog Sorry for barging in but does this mean that psalm will get full-time support, i.e. like how muglug did?

**Comment by danog** [comment]
Yes, and I am open for further support contracts as well, will also update the readme to let the community know (already discussed this with the other maintainers).

**Comment by zmitic** [comment]
@danog 

Thank you and special thanks to Vimeo for doing this. Psalm is just **too good** and I was getting worried about the support, this is a real relief.
